src/Smart.Design.Console/CommandLineParser.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using CommandLine;
+using FluentValidation;
+using Smart.Design.Console.CommandOptions;
+using Smart.Design.Console.Models;
+using Smart.Design.Console.Models.Validations;
+using Smart.Design.Console.UseCases;
+
+namespace Smart.Design.Console
+{
+    public class CommandLineParser
+    {
+        private readonly string[] _args;
+
+        public CommandLineParser(string[] _args)
+        {
+            this._args = _args;
+        }
+
+        public CommandResponse ExecuteCommandWithArguments()
+        {
+            CommandResponse result = Parser.Default.ParseArguments<ListingOptions, GenerateOptions>(_args)
+                .MapResult(
+                    (ListingOptions listOptions) => ListComponentsFromArguments(listOptions),
+                    (GenerateOptions generateOptions) => GeneratePlainHtmlFromArguments(generateOptions), HandleParseError);
+            return result;
+        }
+
+        private  CommandResponse GeneratePlainHtmlFromArguments(GenerateOptions generateOptions)
```

Comment:
Same here for the "FromAruments". I don't think we should name every method in a class with "WithTheParametersGivenToTheConstructor" right ? :)

Reply:
Right, seems fair in that case indeed